aiosmtpd/tests/test_smtp.py

```diff
@@ -1019,11 +1145,11 @@ def test_auth_login_null_credential(self):
             self.assertEqual(response, b"UGFzc3dvcmQA")
             code, response = client.docmd('=')
             assert_auth_success(self, code, response)
-            self.assertEqual(auth_peeker.login, None)
-            self.assertEqual(auth_peeker.password, None)
-            code, response = client.mail("alice@example.com")
-            sess: SMTPSess = self.handler.session
-            self.assertEqual(sess.login_data, b"")
+            assert auth_peeker.mechanism == "LOGIN"
+            assert auth_peeker.login == b""
+            assert auth_peeker.password == b""
+            response = client.mail("alice@example.com")
+            assert response == (250, b"OK")
```

Comment:
Replace the pointless "let's check that login_data doesn't change" test with a test that actually tests that 'protected' commands are accepted.

Also, migrated some tests to pytest's `assert` methodology.